### PR TITLE
remove `map::Op::Nop`

### LINKED ISSUE #24: Get rid of the Nop op in the Map crdt
We use `map::Op::Nop` in some of our tests, but it has no useful semantics, lets get rid of it

### PR DESCRIPTION
**Pull request by chpio**
#BREAKING_CHANGE

Hi,

im (very) new to this (crdts in general but also this crate), so please take a good look at the changes.

fixes #24

### CODE REVIEW

File: test/map.rs [hunk 1]
```diff
@@ -354,11 +366,12 @@ fn test_commute_quickcheck_bug() {
...
-            dot: Dot::new(21, 5),
-            key: 0,
-            op: map::Op::Nop,
-        },
+        // TODO: ???
+        // map::Op::Up {
+        //     dot: Dot::new(21, 5),
+        //     key: 0,
+        //     op: map::Op::Nop,
+        // },
```
Comment (chpio) [inline test/map.rs]: I don't get the purpose of this test, what would be the appropriate replacement for it?

File: test/map.rs [hunk 2]
```diff
@@ -408,19 +421,19 @@ fn test_idempotent_quickcheck_bug2() {
...
-}
-
 #[test]
 fn test_op_exchange_same_as_merge_quickcheck1() {
     let op1 = map::Op::Up {
         dot: Dot::new(38, 4),
         key: 216,
-        op: map::Op::Nop,
+        op: map::Op::Up {
+            dot: Dot::new(38, 1),
```
Comment (chpio) [inline test/map.rs]: the version should be > 4 here?

File: test/map.rs (hunk 1 above)
Comment (davidrusu) [reply test/map.rs]: I wish I would have taken better notes here, this was a bug that came up through quickcheck fuzz testing.

This first Op with nested Nop would create an empty `Map` value under the `0` key and I think that was the crux of this bug, something about the empty Map broke down on merge.

I think the closest replacement would be to replace this Nop with 2 new ops
1. one that adds a value to a nested map under key 0, ie. `{0: {32: 0}}`
2. and a following op that removes the `32` key from the sub map, ie. `{0: {}}`

File: test/map.rs (hunk 2 above)
Comment (davidrusu) [reply test/map.rs]: The top level version vector should always have the highest versions for each actor, the only constraint on these sub-Op versions is that they are smaller or equal to the vector clocks of the parent CRDT.

Comment (davidrusu) [reply test/map.rs]: so `1` is totally fine here

**Comment by davidrusu** [comment]
@chpio this change looks good, is it ready for review?

**Comment by chpio** [comment]
i think so?!

**Comment by davidrusu** [comment]
thanks!

**Pull request merged**